staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -673,6 +674,10 @@ func shouldSyncNode(oldNode, newNode *v1.Node) bool {
 		return true
 	}
 
+	if !equality.Semantic.DeepEqual(oldNode.Status.Addresses, newNode.Status.Addresses) {
+		return true
```

Comment:
Even with `shouldSyncNode` returning true here, `nodeSlicesEqualForLB` compares the node list by name so it may not actually trigger a sync.

Reply:
@andrewsykim but if equal, `updateLoadBalancerHosts` is still called, so it works (and it works when I tested too :smile: )

Reply:
Right, but last time I checked, `updateLoadBalancerHosts` only attempts to retry failed Services. It may have only worked because the controller does a resync of all Services every 100 seconds.

Reply:
Or unless your Service was in a failed state before

Reply:
True I missed this one https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/controllers/service/controller.go#L713 😅

Is adding the same check before not too expensive ? 🤔

Reply:
Which check? Adding all services to `servicesToUpdate`?

Reply:
I think it would be too expensive because we would then sync all LBs even when the node slice is equal.

Reply:
Nop I meant the deepequal on the address on the same if than the `nodeSlicesEqualForLB`

Reply:
Ah.. hmm that one is tricky because not all LB implementations actually check the node address. Which makes me think that the address equal check here may be redundant too. Need to think about this one more.

Reply:
Hum true. But I guess nodes changing IP is very rare so it should rarely happen ?

Reply:
@andrewsykim up? :smile:

Reply:
@andrewsykim double up :smile: